### PR TITLE
Disambiguate "input" and "measurement" (#217)

### LINKED ISSUE #217: Review: 05: Chris Wood: Ambiguous usage of "input" and "measurement"
> There was some inconsistent use of "input" and "measurement", so I started trying to make them uniform in usage throughout. Then I got to the DAF description where we have the function `measurement_to_input_shares`, which accepts values of type `Measurement` as input. To me, I think using `input` consistently here would be and let the reader piece things together. I'd suggest converging on "input," since the values being aggregated over

### PR DESCRIPTION
**Pull request by cjpatton**
Closes #217.

"input" refers to:
* a measurement (we write both "input" and "measurement")
* input to the validity circuit in `Flp` (we write "input" for the plaintext value and "measurement share" for the secret-shared value)
* input share as in `Vdaf` (we are consistent here)
* IDPF to `Idpf` (we are also consistent here)

Resolve the ambiguities by:
* always calling a measurement (i.e., the input to the protocol) a measurement
* always call the input to the validity circuit a measurement

### CODE REVIEW

File: draft-irtf-cfrg-vdaf.md [hunk 1]
```diff
@@ -1817,86 +1819,87 @@ The parameters provided by a concrete FLP are listed in {{flp-param}}.
...
-run on a share of the input and proof, the query-generation algorithm outputs a
-share of the verifier message. Furthermore, the privacy property of the FLP
-system ensures that the verifier message reveals nothing about the input's
-validity. Therefore, to decide if an input is valid, the Aggregators will run
-the query-generation algorithm locally, exchange verifier shares, combine them
-to recover the verifier message, and run the decision algorithm.
+run on a share of the measurement and proof, the query-generation algorithm
+outputs a share of the verifier message. Furthermore, the privacy property of
+the FLP system ensures that the verifier message reveals nothing about the
+measurement's validity. Therefore, to decide if a measurement is valid, the
```
Comment (divergentdave) [inline draft-irtf-cfrg-vdaf.md:1848]: Unrelated to this PR: this is backwards, and contradictory to the definition of `Flp.decide()`.
```suggestion
the FLP system ensures that the verifier message reveals nothing about the measurement
other than whether it is valid. Therefore, to decide if a measurement is valid, the
```

File: draft-irtf-cfrg-vdaf.md [hunk 2]
```diff
@@ -1909,21 +1912,21 @@ also specifies a method of encoding raw measurements as a vector of field
...
+  as a vector of field elements. The return value MUST be of length `MEAS_LEN`.
+
+For some FLPs, the encoded measurement also includes redundant field elements
+that are useful for checking the proof, but which are not needed after the
+proof has been checked. An example is the "integer sum" data type from
+{{CGB17}} in which an integer in range `[0, 2^k)` is encoded as a vector of `k`
+field elements, each representing a bit of the integer (this type is also
+defined in {{prio3sum}}). After consuming this vector, all that is needed is
+the integer it represents. Thus the FLP defines an algorithm for truncating the
+measurement to the length of the aggregated output:
```
Comment (divergentdave) [inline draft-irtf-cfrg-vdaf.md:1924]: For clarity:
```suggestion
encoded measurement to the length of the aggregated output:
```

File: draft-irtf-cfrg-vdaf.md [hunk 3]
```diff
@@ -3235,13 +3241,14 @@ subsections. These methods make use of constants defined in {{poplar1-const}}.
...
 ### Client
 
-The Client's input is an IDPF index, denoted `alpha`. The programmed IDPF values
-are pairs of field elements `(1, k)` where each `k` is chosen at random. This
-random value is used as part of the secure sketching protocol of {{BBCGGI21}},
-Appendix C.4. After evaluating their IDPF key shares on a given sequence of
-candidate prefixes, the sketching protocol is used by the Aggregators to verify
-that they hold shares of a one-hot vector. In addition, for each level of the
-tree, the prover generates random elements `a`, `b`, and `c` and computes
+The Client's measurement includes an IDPF index, denoted `alpha`. The
```
Comment (divergentdave) [inline draft-irtf-cfrg-vdaf.md:3244]: Technically, the  index makes up the entirety of the measurement
```suggestion
The Client's measurement is an IDPF index, denoted `alpha`. The
```

```suggestion
The Client's measurement is interpreted as an IDPF index, denoted `alpha`. The
```

File: poc/flp_generic.py [hunk 4]
```diff
@@ -328,8 +328,8 @@ def test_vec_set_type_param(FlpGeneric, test_vec):
 # GADGETS
 #
 
-def check_gadget_eval(Gadget, inp):
-    if len(inp) != Gadget.ARITY:
+def check_gadget_eval(Gadget, meas):
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply poc/flp_generic.py]: Fair enough. I was thinking something like "function of the encoded measurement", but I think that's confusing. Fixed.

File: draft-irtf-cfrg-vdaf.md [hunk 5]
```diff
@@ -2628,26 +2633,27 @@ The FLP described in the next section extends the proof system of {{BBCGGI19}},
...
 the length-`N` input and `r` is a random field element. The gadget circuit
-`Range2` is the "range-check" polynomial described above, i.e., `Range2(x) = x^2 -
-x`. The idea is that, if `inp` is valid (i.e., each `inp[j]` is in `[0,2)`),
-then the circuit will evaluate to 0 regardless of the value of `r`; but if
-`inp[j]` is not in `[0,2)` for some `j`, the output will be non-zero with high
-probability.
+`Range2` is the "range-check" polynomial described above, i.e., `Range2(x) =
+x^2 - x`. The idea is that, if `meas` is valid (i.e., each `meas[j]` is in
+`[0,2)`), then the circuit will evaluate to 0 regardless of the value of `r`;
+but if `inp[j]` is not in `[0,2)` for some `j`, the output will be non-zero
```
Comment (divergentdave) [inline draft-irtf-cfrg-vdaf.md:2655]: This instance needs to be changed as well
```suggestion
but if `meas[j]` is not in `[0,2)` for some `j`, the output will be non-zero
```

File: draft-irtf-cfrg-vdaf.md [hunk 6]
```diff
@@ -2780,7 +2786,7 @@ In the remainder, we let `[n]` denote the set `{1, ..., n}` for positive integer
 
 #### Proof Generation {#flp-generic-construction-prove}
 
-On input `inp`, `prove_rand`, and `joint_rand`, the proof is computed as
+On measut `meas`, `prove_rand`, and `joint_rand`, the proof is computed as
```
Comment (divergentdave) [inline draft-irtf-cfrg-vdaf.md:2789]: This looks like a find-replace mistake
```suggestion
On input of `meas`, `prove_rand`, and `joint_rand`, the proof is computed as
```

**Review by divergentdave — APPROVED** [review]

**Comment by cjpatton** [comment]
Rebased and squashed.

**Pull request merged**